cognite/client/data_classes/data_modeling/ids.py

```diff
@@ -142,13 +142,28 @@ class DataModelId(VersionedDataModelingId):
     _type = "datamodel"
 
 
+class IdLike(Protocol):
+    space: str
+    external_id: str
+    version: str
```

Comment:
I like this better. The reason I did not like the last one was that it felt semantically wrong. All good for me.